keyvi/include/keyvi/dictionary/dictionary.h

```diff
@@ -32,6 +32,7 @@
 #include "dictionary/fsa/automata.h"
 #include "dictionary/fsa/state_traverser.h"
 #include "dictionary/fsa/traverser_types.h"
+#include "dictionary/fuzzy_match.h"
```

Comment:
I find it a bit confusing: 

- fuzzy_match.h sounds like a derived version of match.h, but that isn't the case.
- it is place top-level, do you think it will be used directly or always via dictionary?

Reply:
what about fuzzy_matcher.h placed in a 'matching' subfolder?

In the long run I would migrate e.g. GetNear there as well. Yours look cleaner than my in-class implementations.

Reply:
Yes, truth be told I was not sure about naming and location either. But I wanted to have it outside of Dictionary class for sure!
Also during testing I've realized that current implementation of `FuzzyCompletion` has a couple of bugs, and as it shares most of code with `FuzzyMatch`, later on it can be reimplemented by just deriving from `FuzzyMatch` and overriding `NextMatch()` method.

Reply:
that sounds like a good idea, but again we have to differ between the matching and the completion case, still a lot of code can be shared.

FWIW 

 - fuzzy completion/matching isn't well tested, I agree
 - it has been written before I rewrote lots of the traversal code (https://github.com/KeyviDev/keyvi/blob/master/keyvi/include/keyvi/dictionary/fsa/traversal/traversal_base.h), which can make a big difference in speed
 - bugs: sure, please open issues for them, I really like to improve the quality of fuzzy matching